server/src/services/group_permissions.service.ts

```diff
@@ -136,6 +137,31 @@ export class GroupPermissionsService {
     } = body;
 
     await getManager().transaction(async (manager) => {
+      //update user group name
+      if (name) {
+        const reservedGroups = ['admin', 'all_users'];
+        if (reservedGroups.includes(groupPermission.group)) {
+          throw new BadRequestException('Cannot update a defualt group name');
+        }
+
+        if (['All Users', 'Admin'].includes(name)) {
```

Comment:
These are just displayed text for group defaults. Converting the name to snake_case and checking against reserved names should do.